### ISSUE TITLE
Classification loss

### ISSUE THREAD

**Issue by ds2268** [facebookresearch/VICRegL#9]
@Adrien987k thank you for the work and the implementation.

I have noticed that you are also using [classification loss](https://github.com/facebookresearch/VICRegL/blob/main/main_vicregl.py#L530), in addition to the VICReg and VICRegL losses. The classification loss is not mentioned in the paper and was also not used in the original VICReg.

On the first hand, it seems a bit off, given that you won't have labels available when pretraining in the wild and it destroys the purpose of "self-supervised" pre-training ?

**Comment by ds2268** [comment]
I have prepared a pull request that should fix the issue (https://github.com/facebookresearch/VICRegL/pull/10). I think that this should not be added to the global loss.

**Comment by ds2268** [comment]
I see that you have used [detach](https://github.com/facebookresearch/VICRegL/blob/main/main_vicregl.py#L481). The learned representation was thus not affected by the gradients coming from the classification loss. Still, this is ugly and the actual loss value does thus not represent the actual VICRegL loss.

**Comment by jinx2018** [comment]
> @Adrien987k thank you for the work and the implementation.
> 
> I have noticed that you are also using [classification loss](https://github.com/facebookresearch/VICRegL/blob/main/main_vicregl.py#L532), in addition to the VICReg and VICRegL losses. The classification loss is not mentioned in the paper and was also not used in the original VICReg.
> 
> On the first hand, it seems a bit off, given that you won't have labels available when pretraining in the wild and it destroys the purpose of "self-supervised" pre-training ?

Hi! have you ever tried to reproduce the models without the classification loss?
